src/key.cpp

```diff
@@ -92,49 +93,55 @@ static int ec_privkey_import_der(const secp256k1_context *ctx,
     size_t privkeylen)
 {
     const unsigned char *end = privkey + privkeylen;
-    int lenb = 0;
-    int len = 0;
+    size_t lenb = 0;
+    size_t len = 0;
     memset(out32, 0, 32);
     /* sequence header */
-    if (end < privkey + 1 || *privkey != 0x30)
+    if (end - privkey < 1 || *privkey != 0x30u)
     {
         return 0;
     }
     privkey++;
     /* sequence length constructor */
-    if (end < privkey + 1 || !(*privkey & 0x80))
+    if (end - privkey < 1 || !(*privkey & 0x80u))
     {
         return 0;
     }
-    lenb = *privkey & ~0x80;
+    lenb = *privkey & ~0x80u;
     privkey++;
     if (lenb < 1 || lenb > 2)
     {
         return 0;
     }
-    if (end < privkey + lenb)
+    if (end - privkey < lenb)
     {
         return 0;
     }
     /* sequence length */
-    len = privkey[lenb - 1] | (lenb > 1 ? privkey[lenb - 2] << 8 : 0);
+    len = privkey[lenb - 1] | (lenb > 1 ? privkey[lenb - 2] << 8 : 0u);
     privkey += lenb;
-    if (end < privkey + len)
+    if (end - privkey < len)
     {
         return 0;
     }
     /* sequence element 0: version number (=1) */
-    if (end < privkey + 3 || privkey[0] != 0x02 || privkey[1] != 0x01 || privkey[2] != 0x01)
+    if (end - privkey < 3 || privkey[0] != 0x02u || privkey[1] != 0x01u || privkey[2] != 0x01u)
     {
         return 0;
     }
     privkey += 3;
     /* sequence element 1: octet string, up to 32 bytes */
-    if (end < privkey + 2 || privkey[0] != 0x04 || privkey[1] > 0x20 || end < privkey + 2 + privkey[1])
+    if (end - privkey < 2 || privkey[0] != 0x04u)
     {
         return 0;
     }
-    memcpy(out32 + 32 - privkey[1], privkey + 2, privkey[1]);
+    size_t oslen = privkey[1];
+    privkey += 2;
+    if (oslen > 32 || end - privkey < oslen)
```

Comment:
nit: not sure i see how making oslen size_t does anything new here but ok.

Reply:
in commit  4a744fb1cf5 we change the type oof `oslen` from `size_t` to `ptrdiff_t`